gdal/port/cpl_swift.cpp

```diff
@@ -190,51 +200,67 @@ bool VSISwiftHandleHelper::AuthV1(CPLString& osStorageURL,
 /*                      CreateAuthV3RequestObject()                     */
 /************************************************************************/
 
-CPLJSONObject VSISwiftHandleHelper::CreateAuthV3RequestObject()
+CPLJSONObject VSISwiftHandleHelper::CreateAuthV3RequestObject(const CPLString& osAuthType)
 {
-    CPLString osUser = CPLGetConfigOption("OS_USERNAME", "");
-    CPLString osPassword = CPLGetConfigOption("OS_PASSWORD", "");
-
-    CPLJSONObject user;
-    user.Add("name", osUser);
-    user.Add("password", osPassword);
-
-    CPLString osUserDomainName = CPLGetConfigOption("OS_USER_DOMAIN_NAME", "");
-    if( ! osUserDomainName.empty() )
-    {
-        CPLJSONObject userDomain;
-        userDomain.Add("name", osUserDomainName);
-        user.Add("domain", userDomain);
-    }
-
-    CPLJSONObject password;
-    password.Add("user", user);
-
     CPLJSONArray methods;
-    methods.Add("password");
-
     CPLJSONObject identity;
-    identity.Add("methods", methods);
-    identity.Add("password", password);
-
     CPLJSONObject scope;
-    CPLString osProjectName = CPLGetConfigOption("OS_PROJECT_NAME", "");
-    if( ! osProjectName.empty() )
+    if ( osAuthType == "v3applicationcredential" )
```

Comment:
a new test case for that method would be appreciated

Reply:
Yep, I'll add one.

Reply:
New test added, just waiting for CI to run.